go-controller/pkg/cluster/master.go

```diff
@@ -74,6 +86,55 @@ func (cluster *OvnClusterController) StartClusterMaster(masterNodeName string) e
 	return nil
 }
 
+func (cluster *OvnClusterController) ReInitClusterNode() error {
+	// setup node
+	var err error
+	var subnet *netIPNet
+
+	nodes, err := cluster.Kube.GetNodes()
+	if err != nil {
+		logrus.Errorf("Error in getting k8s nodes: %v", err)
+		return err
+	}
+	for _, node := range nodes.Items {
+		sub, ok := node.Annotations[OvnHostSubnet]
+		_, subnet, err = net.ParseCIDR(sub)
+		if err != nil {
+			logrus.Errorf("Invalid hostsubnet found for node %s - %v",
+				node.Name, err)
+			return err
+		}
+
+		err = setupOVNNode(node.Name, config.Kubernetes.APIServer,
+			config.Kubernetes.Token)
+		if err != nil {
+			logrus.Errorf("Error setting up OVN node %s - %v", node.Name,
+				err)
+			return err
+		}
+		err = util.RestartOvnController()
+		if err != nil {
+			logrus.Errorf("Error restarting ovncontroller - %v", err)
+			return err
+		}
+		err = ovn.CreateManagementPort(node.Name, subnet.String(),
+			cluster.ClusterIPNet.String(), cluster.ClusterServicesSubnet)
+		if err != nil {
+			logrus.Errorf("Error creating management port on node %s - %v",
+				node.Name, err)
+			return err
+		}
+		if cluster.GatewayInit {
+			err = cluster.initGateway(node.Name,
```

Comment:
initGateway() is local node dependent. You should instead likely have to call util.GatewayInit() directly. The information needed to call  util.GatewayInit should be annotated to node when the original node called it.